cmd/object_get_put_test.go

```diff
@@ -0,0 +1,236 @@
+
+	err = config.InitServer(ctx, modules)
+	require.NoError(t, err)
+
+	fedCancel, err := launchers.LaunchModules(ctx, modules)
+	defer fedCancel()
+	if err != nil {
+		log.Errorln("Failure in fedServeInternal:", err)
+		require.NoError(t, err)
+	}
+
+	desiredURL := param.Server_ExternalWebUrl.GetString() + "/.well-known/openid-configuration"
+	err = server_utils.WaitUntilWorking(ctx, "GET", desiredURL, "director", 200)
+	require.NoError(t, err)
+
+	httpc := http.Client{
+		Transport: config.GetTransport(),
+	}
+	resp, err := httpc.Get(desiredURL)
+	require.NoError(t, err)
+
+	assert.Equal(t, resp.StatusCode, http.StatusOK)
+
+	responseBody, err := io.ReadAll(resp.Body)
+	require.NoError(t, err)
+	expectedResponse := struct {
+		JwksUri string `json:"jwks_uri"`
+	}{}
+	err = json.Unmarshal(responseBody, &expectedResponse)
+	require.NoError(t, err)
+
+	assert.NotEmpty(t, expectedResponse.JwksUri)
+	//////////////////////////////////////////////////////////////////////////////////////////
+	t.Run("testObjectPutAndGet", func(t *testing.T) {
+		testFileContent := "test file content"
+		// Create the temporary file to upload
+		tempFile, err := os.CreateTemp(t.TempDir(), "test")
+		assert.NoError(t, err, "Error creating temp file")
+		defer os.Remove(tempFile.Name())
+		_, err = tempFile.WriteString(testFileContent)
+		assert.NoError(t, err, "Error writing to temp file")
+		tempFile.Close()
+
+		// Create a token file
+		token, err := generateFileTestScitoken()
+		assert.NoError(t, err)
+		tempToken, err := os.CreateTemp(t.TempDir(), "token")
+		assert.NoError(t, err, "Error creating temp token file")
+		defer os.Remove(tempToken.Name())
+		_, err = tempToken.WriteString(token)
+		assert.NoError(t, err, "Error writing to temp token file")
+		tempToken.Close()
+		// Disable progress bars to not reuse the same mpb instance
+		viper.Set("Logging.DisableProgressBars", true)
+
+		// Set path for object to upload/download
+		tempPath := tempFile.Name()
+		fileName := filepath.Base(tempPath)
+		uploadURL := "osdf:///test/" + fileName
+
+		// Upload the file with PUT
+		rootCmd.SetArgs([]string{"object", "put", tempFile.Name(), uploadURL, "-d", "-t", tempToken.Name()})
+		err = rootCmd.Execute()
+		assert.NoError(t, err, "Failed to run pelican object put")
+
+		out, err := io.ReadAll(output)
+		assert.NoError(t, err)
+		// Confirm we're uploading size we are expecting
+		assert.Contains(t, string(out), "Uploaded bytes: 17")
+
+		// Download that same file with GET
+		rootCmd.SetArgs([]string{"object", "get", uploadURL, t.TempDir(), "-t", tempToken.Name(), "-c", param.Origin_Url.GetString(), "-d"})
+		err = rootCmd.Execute()
+		assert.NoError(t, err, "Failed to run pelican object get")
+
+		out, err = io.ReadAll(output)
+		assert.NoError(t, err)
+		// Confirm we download same amount of bytes as upload
+		assert.Contains(t, string(out), "Downloaded bytes: 17")
+	})
+	// cleanup
+	os.RemoveAll(tmpPath)
+	os.RemoveAll(originDir)
+
+	cancel()
+	fedCancel()
```

Comment:
You've already called `defer cancel ()` around line 98 and `defer fedCancel()` around line 153, so you don't want to call them here again to avoid duplicated calls to cancel functions